File: queues/lifo/lifo.rb
# create an example of a lifo queue (last in first out)

class LifoQueue
  def initialize
    @queue = []
  end

  def add(item)
    @queue.push(item)
  end

  def remove
    @queue.pop
  end

  def print_queue
    puts "queue: #{@queue}"
  end
end


lifo_queue = LifoQueue.new
puts "adding 1"
lifo_queue.add(1)
lifo_queue.print_queue
puts "adding 2"
lifo_queue.add(2)
lifo_queue.print_queue
puts "adding 3"
lifo_queue.add(3)
lifo_queue.print_queue

puts "removing a value, expects 3, got: #{lifo_queue.remove}"
lifo_queue.print_queue
puts "removing a value, expects 2, got: #{lifo_queue.remove}"
lifo_queue.print_queue
puts "removing a value, expects 1, got: #{lifo_queue.remove}"
lifo_queue.print_queue

File: queues/lifo/lifo.js
// create an example of a lifo queue (last in first out) in javascript

class LifoQueue {
  constructor() {
    this.queue = [];
  }

  add(item) {
    this.queue.push(item);
  }

  remove() {
    return this.queue.pop();
  }

  printQueue() {
    console.log(this.queue);
  }
}

const lifoQueue = new LifoQueue();
lifoQueue.add(1);
console.log("queue should be [1]");
lifoQueue.printQueue();
lifoQueue.add(2);
console.log("queue should be [1, 2]");
lifoQueue.printQueue();
lifoQueue.add(3);
console.log("queue should be [1, 2, 3]");
lifoQueue.printQueue();
console.log("remove should be 3");
console.log(lifoQueue.remove());
console.log("queue should be [1, 2]");
console.log(lifoQueue.queue);
